crates/framework/src/natives/mos_stdlib/object_extension/mod.rs

```diff
@@ -0,0 +1,174 @@
+        let object_data = self.object_data.borrow();
+        //TODO need to check the transfers
+        // if let Some(object) = object_data.transfers.get(&object_id) {
+        //     return Ok(Some(object.clone()));
+        // }
+        if object_data.removed_object.contains(&object_id) {
+            return Ok(None);
+        }
+        if object_data.new_object_ids.contains(&object_id) {
+            return Ok(None);
+        }
+        let object = self.resolver.resolve_object(object_id).map_err(|e| {
+            PartialVMError::new(StatusCode::STORAGE_ERROR).with_message(format!("{:?}", e))
+        })?;
+        Ok(object)
+    }
+
+    pub fn delete_id(&self, object_id: ObjectID) -> Result<(), PartialVMError> {
+        let mut object_data = self.object_data.borrow_mut();
+        object_data.removed_object.insert(object_id);
+        Ok(())
+    }
+
+    pub(crate) fn transfer(
+        &self,
+        owner: Owner,
+        ty: Type,
+        tag: StructTag,
+        layout: MoveStructLayout,
+        obj: Value,
+    ) -> Result<TransferResult, PartialVMError> {
+        let mut object_data = self.object_data.borrow_mut();
+        let object_id: ObjectID = get_object_id(obj.copy_value()?)?
+            .value_as::<AccountAddress>()?
+            .into();
+        object_data
+            .transfers
+            .insert(object_id, ObjectInfo::new(owner, ty, tag, layout, obj));
+        //TODO return the correct result
+        Ok(TransferResult::New)
+    }
+
+    pub fn into_change_set(self) -> Result<ObjectChangeSet, PartialVMError> {
+        let object_data = self.object_data.into_inner();
+        //TODO process change set.
+        Ok(ObjectChangeSet {
+            new_objects: object_data.transfers,
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct ObjectInfo {
+    pub owner: Owner,
+    pub ty: Type,
+    pub tag: StructTag,
+    //TODO should contains layout?
+    pub layout: MoveStructLayout,
+    pub value: Value,
+}
+
+impl ObjectInfo {
+    pub fn new(
+        owner: Owner,
+        ty: Type,
+        tag: StructTag,
+        layout: MoveStructLayout,
+        value: Value,
+    ) -> Self {
+        Self {
+            owner,
+            ty,
+            tag,
+            layout,
+            value,
+        }
+    }
+}
+
+impl std::fmt::Display for ObjectInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "ObjectInfo({:?}, {:?}, {:?}, {:?})",
+            self.owner, self.ty, self.tag, self.value
+        )
+    }
+}
+
+/// A structure representing mutable data of the NativeTableContext. This is in a RefCell
+/// of the overall context so we can mutate while still accessing the overall context.
+#[derive(Default)]
+struct ObjectData {
+    new_object_ids: BTreeSet<ObjectID>,
+    removed_object: BTreeSet<ObjectID>,
+    transfers: BTreeMap<ObjectID, ObjectInfo>,
+}
```

Comment:
new_object_ids -> new_ids ?
removed_object -> removed_ids ?
transfers -> transferred_id_infos ?

Reply:
Ignore it, the removed_object field has not been used and does not implements.